main/ruletester/ruletester.go

```diff
@@ -66,22 +62,21 @@ func ReadMetricsFile(file string) ([]string, error) {
 		return nil, err
 	}
 
-	result := make(map[string]int)
+	// Read the data with a zlib reader
 	r, err := zlib.NewReader(bytes.NewBuffer(data))
+	defer r.Close()
 	if err != nil {
-		panic("Problem with zlib compressed data")
+		return nil, fmt.Errorf("Problem with zlib compressed data: %s", err.Error())
 	}
-	b := new(bytes.Buffer)
-	io.Copy(b, r)
-	r.Close()
-	d := gob.NewDecoder(b)
 
-	err = d.Decode(&result)
+	// Store the result of the decode in this map:
+	result := map[string]int{}
+	err = gob.NewDecoder(r).Decode(&result)
 	if err != nil {
 		return nil, err
 	}
 
-	strings := []string{}
+	strings := make([]string, 0, len(result))
```

Comment:
I like `[]string{}` since it's pretty concise. `make()` with capacity is a bit of a micro-optimization, but I'm ok with it if we already know how big the slice is going to be.